beeware/briefcase
# Added code for supporting arch system package

## Testing instructions:

If you're on an Arch Linux system, or any derivative of an Arch system (e.g., Manjaro), you can test this branch with:
```
mkdir bee-test
cd bee-test
python3 -m venv venv
source venv/bin/activate
pip install git+https://github.com/proneon267/briefcase.git@patch-1
briefcase new --template https://github.com/beeware/briefcase-template --template-branch arch
```
Accept all the default values from the template, then run:
```
cd helloworld
briefcase run
```
You should get a running Toga GTK app. Then, run:
```
briefcase package
```
This should generate a `dist/helloworld-0.0.1-x86_64.pkg.tar.zst` file, which you can install with `pacman -U dist/helloworld-0.0.1-x86_64.pkg.tar.zst`

If you have any problems building, running or installing this package, or the package that is generated doesn't match your expectations as an Arch* user, let us know in the comments below, including details of the specific distro that you are using for testing.

---


For creating arch packages, we need to build a `PKGBUILD` file which will be used by `makepkg` utility to create the `.pkg.tar.zst` file. 
A sample `PKGBUILD` file:
```
# Maintainer: Your Name <youremail@domain.com>
pkgname=NAME
pkgver=VERSION
pkgrel=1
epoch=
pkgdesc=""
arch=()
url=""
license=('GPL')
groups=()
depends=()
makedepends=()
checkdepends=()
optdepends=()
provides=()
conflicts=()
replaces=()
backup=()
options=()
install=
changelog=
source=("$pkgname-$pkgver.tar.gz"
        "$pkgname-$pkgver.patch")
noextract=()
md5sums=()
validpgpkeys=()

prepare() {
	cd "$pkgname-$pkgver"
	patch -p1 -i "$srcdir/$pkgname-$pkgver.patch"
}

build() {
	cd "$pkgname-$pkgver"
	./configure --prefix=/usr
	make
}

check() {
	cd "$pkgname-$pkgver"
	make -k check
}

package() {
	cd "$pkgname-$pkgver"
	make DESTDIR="$pkgdir/" install
}
```

There are many parameters but we do not need to use them all. As you can see among all these functions, only `package()` is necessary, others are optional.
`makepkg` doesn't necessarily need to compile the stub binary. Since, `briefcase build` does the stub binary compilation, we can skip this step.
A simplified and tested `PKGBUILD` file:
```
pkgname=helloworld
pkgver=0.0.1
pkgrel=1
pkgdesc="This is a package"
arch=('x86_64')
url="https://github.com"
license=('GPL')
depends=('python' 'gobject-introspection' 'gobject-introspection-runtime' 'glibc' 'gtk3')
source=("$pkgname-$pkgver.tar.gz")
md5sums=('SKIP')
changelog=CHANGELOG
package() {
  cp -r "$srcdir/$pkgname-$pkgver"/usr/ "$pkgdir"/usr/
}
```


I will add more details later on. 
This PR relates to an issue: #1064 
## PR Checklist:

- [ ] All new features have been documented
